File: Level 2/Week04/Day22/hhs/1.java
class Solution {
    public int getMaxDivisor(int n) {
        if (n == 1) return 0;

        int div = 1;
        for (int i = 2; i * i <= n; i++) {
            if (n % i == 0) {
                div = i;
                if (n / div <= 10000000) return n / div;
            }
        }

        return div;
    }

    public int[] solution(long begin, long end) {
        int[] answer = new int[(int) (end - begin) + 1];

        int index = 0;

        for (int i = (int) begin; i <= end; i++) {
            answer[index++] = getMaxDivisor(i);
        }

        return answer;
    }
}

File: Level 2/Week04/Day23/hhs/1.kt
package pro_42888

class Solution2 {

    fun solution(record: Array<String>): Array<String> {

        val map = mutableMapOf<String, String>()
        val answer = ArrayList<String>()

        for (r in record) {
            val split = r.split(" ")

            when (split[0]) {
                "Enter", "Change" -> map += split[1] to split[2]
            }
        }

        for (r in record) {
            val split = r.split(" ")

            when (split[0]) {
                "Enter" -> answer.add("${map[split[1]]}님이 들어왔습니다.")
                "Leave" -> answer.add("${map[split[1]]}님이 나갔습니다.")
            }
        }

        return answer.toTypedArray()
    }
}